src/test/java/software/amazon/cloudformation/resource/ValidatorTest.java

```diff
@@ -322,6 +322,45 @@ public void validateDefinition_invalidHandlerSection_shouldThrow() {
             .withNoCause().withMessage("#/handlers/read: required key [permissions] not found");
     }
 
+    @Test
+    public void validateDefinition_validReplacementStrategy_shouldNotThrow() {
```

Comment:
parameterize this one too to include both create/delete and delete/create

Reply:
Good point. Parameterized all the tests